ui/tests/e2e/login.test.ts

```diff
@@ -0,0 +1,56 @@
+import puppeteer from 'puppeteer'
+import { LOGIN_URL, OVERVIEW_URL, PUPPETEER_CONFIG } from './test_config'
+
+describe('Login', () => {
+  let browser
+  beforeAll(async () => {
+    browser = await puppeteer.launch(PUPPETEER_CONFIG)
+  })
+
+  afterAll(() => {
+    browser.close()
+  })
+
+  it(
+    'should login fail by incorrect password',
+    async () => {
+      const page = await browser.newPage()
+      await page.goto(LOGIN_URL)
+
+      const pwdInput = await page.waitForSelector('input#tidb_signin_password')
+      await pwdInput.type('any')
+      await page.click('button#signin_btn')
+
+      const errorContainer = await page.waitForSelector('div.has-error')
+      const containsPwdInput = await errorContainer.$eval(
+        'input#tidb_signin_password',
+        (el) => (el ? true : false)
+      )
+      expect(containsPwdInput).toBe(true)
+      const errorContent = await errorContainer.$eval(
+        '.ant-form-explain',
+        (el) => el.textContent
+      )
```

Comment:
I have a solution for this one: https://www.npmjs.com/package/expect-puppeteer

Use `await expect(page).toMatch('TiDB authentication failed')`. No more magic css selectors.

Other utilities provided by this library is also pretty handy.. like

`await expect(page).toClick('button#signin_btn')`

Reply:
OK, let me try it soon.

Reply:
> Remember to open subsequent PRs to drop the code that heavily rely on ant design.

Create an issue https://github.com/pingcap-incubator/tidb-dashboard/issues/287 to refine the subsequent work, will do it this week if have time.